File: src/Services/Bots/Bots.Domain/Entities/Bot.cs
namespace Bots.Domain.Entities
{
    public class Bot
    {
        public int Id { get; set; }
        public string IP { get; set; }
        public string Platform { get; set; }
        public string Status { get; set; }

        public int BotnetId { get; set; }
    }
}

File: src/BotnetWebApp/botnetwebapp/src/components/Bot.js
import React from "react";

function Bot({ bot, deleteBot }) {
  const handleDelete = () => {
    deleteBot(bot);
  };

  return (
    <tr>
      <td>{bot.id}</td>
      <td placeholder="Bot IP">{bot.ip}</td>
      <td>{bot.platform}</td>
      <BotStatus bot={bot} />
      <td>
        <button onClick={handleDelete}>Delete?</button>
      </td>
    </tr>
  );
}

function BotStatus({ bot }) {
  if (bot.status === "Working") {
    return <td style={{ color: "darkgreen" }}>{bot.status}</td>;
  } else {
    return <td style={{ color: "darkred" }}>{bot.status}</td>;
  }
}

export default Bot;
